### ISSUE TITLE
[SPMD] prompt-logprobs precompile hangs multi-host startup / redundant when prompt_logprobs unused

### ISSUE THREAD

**Issue by lokic233** [vllm-project/tpu-inference#3455]
### Summary

On the bare-SPMD multi-host backend (#3262), the unconditional prompt-logprobs precompile in `TPUModelRunner`'s `_precompile_gather_logprobs()` (`tpu_inference/runner/compilation_manager.py`) can **hang engine startup at a compile-time rendezvous**, and is redundant work whenever `prompt_logprobs` is never requested at runtime. There is currently no gate to skip it.

### Environment

- TPU v7x (Ironwood), 4×4×4, 64 chips, single slice
- Multi-host backend = `spmd` (bare `jax.distributed`, no Ray) — the backend added in #3262
- Collocated GRPO rollout (Gemma-class 2B), `return_logprobs=False` (only sampled-token logprobs consumed downstream)

### Mechanism

`_precompile_gather_logprobs()` runs an unconditional `compile_only` loop over prompt-token-padding buckets for `compute_and_gather_prompt_logprobs` (guarded only by the hardcoded `MAX_PRECOMPILE_PROMPT_TOKENS = 1024` upper cap):

```python
MAX_PRECOMPILE_PROMPT_TOKENS = 1024
for num_tokens in self.runner.num_tokens_paddings:
    if num_tokens > MAX_PRECOMPILE_PROMPT_TOKENS:
        continue
    ...
    self._run_compilation("worker{rank} compute_and_gather_prompt_logprobs", ...)
```

Under the bare-SPMD backend, a `jit` over the multi-host mesh is itself a **collective/distributed action**: all hosts must enter the *same* compilation with the *same* trip count, or they deadlock at the compile-time rendezvous. In our runs the engine freezes at `Precompile worker0 compute_and_gather_prompt_logprobs` and never emits a step. (This is distinct from the sampled-token `compute_and_gather_logprobs` precompile, which is a separate earlier block and is fine.)

Separately, even when it does not hang (e.g. Ray/single-host), this loop is **pure startup overhead** for any deployment that never requests `prompt_logprobs` — the compiled graph is off the execution path when `return_logprobs=False`.

### Proposed fix

Add an env/config gate (default **off** → zero behavior change for existing users) that skips the prompt-logprobs precompile loop. PR to follow.

### Correctness

`prompt_logprobs` is not requested under stock `return_logprobs=False` (only sampled-token logprobs are read), so skipping its *precompile* leaves the graph off the execution path — no cold-compile / missing-cache assertion mid-decode. Loss-neutral.
